esp/services/ws_machine/ws_machineService.cpp

```diff
@@ -2490,42 +2490,49 @@ void Cws_machineEx::readOtherComponentUsageReq(const char* name, const char* typ
     if (!strieq(type, eqDali) && !strieq(type, eqEclAgent) && !strieq(type, eqSashaServer))
         throw MakeStringException(ECLWATCH_CANNOT_GET_ENV_INFO, "Component usage function is not supported for %s", type);
 
-    Owned<IPropertyTree> envRoot = &constEnv->getPTree();
-    VStringBuffer xpath("Software/%s[@name='%s']/Instance/@computer", type, name);
-    const char* computer = envRoot->queryProp(xpath);
-    if (isEmptyString(computer))
-        throw MakeStringException(ECLWATCH_CANNOT_GET_ENV_INFO, "Failed to get %s", xpath.str());
-
-    Owned<IPropertyTree> machineReq = createMachineUsageReq(constEnv, computer);
-
-    IPropertyTree* envDirectories = envRoot->queryPropTree("Software/Directories");
-    Owned<IPropertyTree> logFolder = createDiskUsageReq(envDirectories, "log", type, name);
-    if (logFolder)
-        machineReq->addPropTree(logFolder->queryName(), LINK(logFolder));
+    Owned<IPropertyTree> componentReq = createPTree("Component");
+    componentReq->addProp("@name", name);
+    componentReq->addProp("@type", type);
```

Comment:
minor/not new: setProp for attributes is clearer (and equivalent), since it's not possible to add the same attribute twice.